### PR TITLE
[docs] add new roadmap items

### PR DESCRIPTION
**Pull request by odow**
As discussed on today's nonlinear call.

### CODE REVIEW

File: docs/src/developers/roadmap.md [hunk 1]
```diff
@@ -8,6 +8,21 @@ Most of these issues will require changes to both JuMP and MathOptInterface, and
 are non-trivial in their implementation. They are in no particular order, but
 represent broad themes that we see as areas in which JuMP could be improved.
 
+ - Support nonlinear expressions with vector-valued inputs and outputs. There
```
Comment (odow) [inline docs/src/developers/roadmap.md:11]: cc @ccoffrin

File: docs/src/developers/roadmap.md [hunk 2]
```diff
@@ -8,6 +8,21 @@ Most of these issues will require changes to both JuMP and MathOptInterface, and
...
+   are a few related components:
+    - Representing terms like `log(det(X))` as necessary for `Convex.jl`
+    - Automatic differentiation of terms with vector inputs and outputs
+    - User-defined functions with vector--as opposed to scalar--inputs, which is
+      particularly useful for optimal control problems
+    - User-defined functions with vector outputs, avoiding the need for
+      [User-defined operators with vector outputs](@ref)
+ - Add support for modeling with SI units. The [UnitJuMP.jl](https://github.com/trulsf/UnitJuMP.jl)
```
Comment (odow) [inline docs/src/developers/roadmap.md:19]: cc @trulsf

**Comment by pulsipher** [comment]
Would something to address #3498 be worth considering?

**Comment by odow** [comment]
I guess #3498 is in the same vein as #3438. We don't really have a concrete suggestion for what to do.

**Comment by joaquimg** [comment]
I know DCP is coming, but it certainly deserves a bullet.

**Comment by odow** [comment]
I added ```Representing terms like `log(det(X))` as necessary for `Convex.jl` ```, which is basically all we need for DCP.

**Comment by odow** [comment]
Merging for now. Can always edit in future.

**Pull request merged**